client/components/chart/index.jsx

```diff
@@ -84,10 +85,13 @@ class Chart extends React.Component {
 		const width = isTouch && clientWidth <= 0 ? 350 : clientWidth; // mobile safari bug with zero width
 		const maxBars = Math.floor( width / minWidth );
 
-		this.setState( { maxBars, width }, () =>
+		const updatedData = this.calculateUpdatedData(
 			// this may get called either directly or as a resize event callback
-			this.updateData( props instanceof Event ? this.props : props )
+			props instanceof Event ? this.props : props,
+			maxBars
```

Comment:
We'd want to use `{ maxBars }` here to match the `calculateUpdatedData`'s param signature.

Reply:
Looking at this `resize`, `updateData` and `calculateUpdatedData` family of methods, it seems that parts of the code is legacy noise.

The first parameter to `calculateUpdatedData` turns out to always be `this.props`. And there are several overly complicated code paths that always end with the same result 🙂 

What if the second parameter `calculateUpdatedData` didn't default to `this.state`, but would always be passed explicitly? I think that would lead to a better code, too.

Finally, what if we passed `data` and `maxBar` parameters instead of `props` and `state`?

Reply:
Excellent catch, @jsnmoon!

Yes, you're right, @jsnajdr, the method signatures could really use some work here. Let me take care of that.

Reply:
Done! I believe this functionally does the same thing, and is much simpler.